UnicornRender/video/include/unicorn/video/Mesh.hpp

```diff
@@ -74,17 +77,28 @@ class Mesh : public Transform
     *
     * @return mesh material data
     */
-    UNICORN_EXPORT Material const& GetMaterial() const;
+    UNICORN_EXPORT std::shared_ptr<Material> GetMaterial() const;
 
     /** @brief Event triggered when material is changed */
     wink::signal<wink::slot<void()>> MaterialUpdated;
 
     /** @brief Event triggered when vertices are changed */
     wink::signal<wink::slot<void()>> VerticesUpdated;
+
+    std::vector<std::string> m_diffuse;
```

Comment:
Why prefix the public variable? also what's up with these vectors of strings? doxygen?